aws/resource_aws_backup_plan.go

```diff
@@ -151,8 +151,14 @@ func resourceAwsBackupPlanRead(d *schema.ResourceData, meta interface{}) error {
 
 		if r.Lifecycle != nil {
 			l := make(map[string]interface{})
-			l["delete_after"] = aws.Int64Value(r.Lifecycle.DeleteAfterDays)
-			l["cold_storage_after"] = aws.Int64Value(r.Lifecycle.MoveToColdStorageAfterDays)
+
+			if v := r.Lifecycle.DeleteAfterDays; v != nil {
```

Comment:
Nit: The `nil` checks here and below are extraneous because [`aws.Int64Value()`](https://docs.aws.amazon.com/sdk-for-go/api/aws/#Int64Value) will automatically convert `nil` to `0` and `TypeInt` defaults to `0`.